source/common/common/thread.cc

```diff
@@ -21,7 +26,13 @@ Thread::Thread(std::function<void()> thread_routine) : thread_routine_(thread_ro
   UNREFERENCED_PARAMETER(rc);
 }
 
-int32_t Thread::currentThreadId() { return syscall(SYS_gettid); }
+int32_t Thread::currentThreadId() {
+#ifdef __linux__
+  return syscall(SYS_gettid);
+#elif defined(__APPLE__)
+  return pthread_mach_thread_np(pthread_self());
```

Comment:
The current method has the advantage of returning 32 bit ints. After some experimentation the values from the current method end up being repeatable across executions. Given that `currentThreadId` is used as part of a seed for a random number generator in `source/common/runtime/runtime_impl.h`, I'll switch to returning the low order bits of `pthread_threadid_np`'s thread id.